server/src/test/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateServiceTests.java

```diff
@@ -301,12 +301,14 @@ public void testAddIndexTemplateV2() throws Exception {
         assertNotNull(state.metadata().templatesV2().get("foo"));
         assertTemplatesEqual(state.metadata().templatesV2().get("foo"), template);
 
+        IndexTemplateV2 newTemplate = IndexTemplateV2Tests.randomInstance();
```

Comment:
It's suuuuper unlikely, but this might be better as:

```java
IndexTemplateV2 newTemplate = randomValueOtherThanMany(t -> Objects.equals(template.priority(), t.priority(), () -> IndexTemplateV2Tests.randomInstance());
```

To prevent generating a new template that accidentally has the same index patterns and priority (highly unlikely but not impossible)

Reply:
Good idea, changed